### ISSUE TITLE
Sending email freezes - deadlock

### ISSUE THREAD

**Issue by Valeriy1991** [lukencode/FluentEmail#104]
Hello!

This is a very **danger**, because it's can result to **deadlock** (since async code is blocked):
https://github.com/lukencode/FluentEmail/blob/d2ad499e611cc94f9bc68e90731a229d25fa3dcf/src/Senders/FluentEmail.Smtp/SmtpSender.cs#L43

More info: https://blog.stephencleary.com/2012/07/dont-block-on-async-code.html (also thanks to @cosmmm for a hint about blocking asynchronous operations)

#### My environments:
- [Mailhog](https://github.com/mailhog/MailHog) as fake of SMTP-server
- VS 2017 15.5.1
- .NET Framework 4.6.2
- FluentEmail.Core - v 2.3.1 -> 2.4.0
- FluentEmail.Smtp - v 2.3.0 -> 2.4.1

#### Steps to reproduce deadlock:
1. Create new "ASP.NET Web Application (.NET Framework)" with 4.6.2
1. Select WebAPI checkbox or MVC + WebAPI
1. Install `FluentEmail.Core` and `FluentEmail.Smtp`
1. Create `EmailController`:
```csharp
   public class EmailController : ApiController
    {
        private SmtpClient CreateSmtpClient()
        {
            return new SmtpClient("my.mailhog.server", 1025)
            {
                DeliveryMethod = SmtpDeliveryMethod.Network,
                EnableSsl = false,
                UseDefaultCredentials = true
            };
        }

        [HttpPost]
        [Route("api/email/send1")]
        public async Task<IHttpActionResult> Send1()
        {
            using (var smtpClient = CreateSmtpClient())
            {
                var emailData = CreateEmailData();
                var mailMessage = new MailMessage()
                {
                    Body = emailData.Body,
                    Subject = emailData.Subject,
                    IsBodyHtml = emailData.IsHtml,
                    From = new MailAddress(emailData.FromAddress.EmailAddress, emailData.FromAddress.Name)
                };
                emailData.ToAddresses.ForEach(address =>
                {
                    mailMessage.To.Add(new MailAddress(address.EmailAddress, address.Name));
                });

                // (1) Return Task with status "WaitingForActivation"
                //var task = smtpClient.SendMailAsync(mailMessage);

                // (2) As in FluentEmail.Smtp - is not work:
                //smtpClient.SendMailAsync(mailMessage).GetAwaiter().GetResult();

                // (3) Sync call - it's work:
                //smtpClient.Send(mailMessage);

                // (4) Async call - it's work:
                await smtpClient.SendMailAsync(mailMessage);

                return Ok();
            }
        }

        [HttpPost]
        [Route("api/email/send2")]
        public IHttpActionResult Send2()
        {
            using (var smtpClient = CreateSmtpClient())
            {
                var emailSender = new SmtpSender(smtpClient)
                {
                    // For v 2.3.0
                    //UseSsl = smtpClient.EnableSsl
                };

                var email = new Email()
                {
                    Data = CreateEmailData()
                };
                // (5) Deadlock:
                //emailSender.Send(email);

                // (6) Async call as sync - it's work:
                var sendResponse = Task.Run(() => emailSender.SendAsync(email)).Result;

                return Ok();
            }
        }
        private static EmailData CreateEmailData()
        {
            var emailData = new EmailData()
            {
                Body = "body",
                Subject = "Subject - TEST 1",
                IsHtml = true,
                ToAddresses = new List<Address>()
                {
                    new Address("null@null.null")
                },
                FromAddress = new Address("Mailhog-TEST@my-company.info")
            };
            return emailData;
        }
    }
```
5. Run application and try send POST-requests to `/api/email/send1` and `/api/email/send2` methods with comment/uncomment lines (1), (2), (3), (4), (5) or (6):
  - in cases (1), (2) and (5) deadlock was occured, but email was sent successfully (I can see it in my Mailhog web UI);
  - in cases (3), (4) and (6) email was sent successfully without any deadlocks.

### Possible solutions:
1. Async in everywhere - just remove the sync-method `SendResponse Send(IFluentEmail email, CancellationToken? token = null)`
2. Try fix deadlock problem

@lukencode, what do you think about it?

**Comment by olibanjoli** [comment]
I fixed this issue in #108 commit 79e5ac4.

**Comment by bjcull** [comment]
Fixed in Nuget version 2.5.1.

Use `Send` for sync, and `await SendAsync` for async.
